main/webapp/modules/core/scripts/index/parser-interfaces/xml-parser-ui.js

```diff
@@ -262,7 +262,7 @@ Refine.XmlParserUI.prototype._updatePreview = function() {
   var self = this;
   
   if(!self._recordElementSelected) {
-    alert("No record element selected! Please select one first.");
+    window.alert($.i18n('core-index-import/warning-record-path'));
```

Comment:
It's still not clear to me why we have to duplicate the existing check, where the same message is used?

Reply:
The existing check works only when the user clicks the `Create Project` button. The `Update Preview` gets stuck if the user does not select one and clicks on any other configuration option. Once that happens, the user cannot do anything (and this happens before Create Project), there isn't any check for this right now. Maybe we can remove the one in `Create Project` because the check in `Update Preview` will make sure that a Record Element is always selected?

Reply:
I think this happens because once `_updatePreview` calls `._getOptions` where the recordPath is set to a defualt value when there isn't already one, which in this case is an empty array. Once the flow of control reaches `XMLImportUtilities`, an error ArrayIndexOutOfBoundsException is generated in `findRecord` because `recordPath` is 0 and we cannot access its 0th index.

Reply:
It looks like we could potentially remove the boolean you introduce and do the same check as for the other `window.alert`, no?

Reply:
Yes okay, I wll do that then. Should I also remove the check from confirmCreateProject then?

Reply:
If that is redundant, sure.